src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -228,7 +228,7 @@ protected void SetInputScope(InputTypes types)
 		{
 			if (_textBoxView != null)
 			{
-				_textBoxView.InputType = types;
+				_textBoxView.SetRawInputType(types);
```

Comment:
This is likely to be a breaking change for application current expecting uno to do the validation. This new behavior is likely to be the right one, though. We should be placing this behind a feature flag.

Reply:
@jeromelaban To be more clear about the current behavior, we indeed was mapping to an Android API that does validation, but *somehow*, Uno managed to skip/override the native validation(probably [this](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Overrides.Android.cs)? not sure), but anyway the result was the cursor moving to the beginning after every invalid character.

So I don't expect anyone to be relying on the current behavior.

Reply:
SIP is Soft Input Panel (the on-screen keyboard). The old behavior was that keys outside the input type would not be entered at all if using a physical keyboard (and the cursor would not more).

Reply:
@jeromelaban Thanks, got it and addressed the feedback.